crates/ra_syntax/src/ast/extensions.rs

```diff
@@ -91,6 +91,7 @@ impl ast::Attr {
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub enum PathSegmentKind {
     Name(ast::NameRef),
+    Angle { type_ref: Option<ast::TypeRef>, trait_ref: Option<ast::PathType> },
```

Comment:
Hm, should we perhaps introduce different variants for `<Type>` and `<Type as Trait>` flavors?  I am also not too thrilled by `Angle` name, but `UFCS` is worse, and `Qualified` (used by "official" grammar) is outright confusing. 

Perhaps just call it `Type`? That's what it is: `<>` is a way to explicitly switch from expr grammar to types grammar.

Reply:
Hmm yeah I'm fine with `Type`.